modules/login/php/ajaxlogin.class.inc

```diff
@@ -0,0 +1,405 @@
+            }
+        }
+
+        $DB       = \Database::singleton();
+        $name     = htmlspecialchars($values['firstname'], ENT_QUOTES);
+        $lastname = htmlspecialchars($values['lastname'], ENT_QUOTES);
+        $from     = htmlspecialchars($values['email'], ENT_QUOTES);
+        $site     = $values['site'];
+        $fullname = $name.' '.$lastname;
+
+        $vals = array(
+                 'UserID'           => $from,
+                 'Real_name'        => $fullname,
+                 'First_name'       => $name,
+                 'Last_name'        => $lastname,
+                 'Pending_approval' => 'Y',
+                 'Email'            => $from,
+                );
+
+        // Check email address' uniqueness
+        $result = $DB->pselectOne(
+            "SELECT COUNT(*) FROM users WHERE Email = :VEmail",
+            array('VEmail' => $from)
+        );
+
+        $user_id = null;
+
+        if ($result == 0) {
+            // insert into DB only if email address doesn't exist
+            $DB->insert('users', $vals);
+            // Get the ID of the new user and insert into user_psc_rel
+            $user_id = $DB->getLastInsertId();
+
+            $DB->insert(
+                'user_psc_rel',
+                array(
+                 'UserID'   => $user_id,
+                 'CenterID' => $site,
+                )
+            );
+
+            if ($values['examiner'] == 'true') {
+                $rad = 0;
+                if ($values['radiologist'] == 'true') {
+                    $rad = 1;
+                }
+                //insert in DB as inactive until account approved
+                $examinerID = $DB->pselect(
+                    "SELECT e.examinerID
+                     FROM examiners e
+                     WHERE e.full_name=:fn",
+                    array("fn" => $fullname)
+                );
+
+                // If examiner not in table add him
+                if (empty($examinerID)) {
+                    $DB->insert(
+                        'examiners',
+                        array(
+                         'full_name'   => $fullname,
+                         'radiologist' => $rad,
+                         'userID'      => $user_id,
+                        )
+                    );
+                    $examinerID = $DB->getLastInsertID();
+                    $DB->insert(
+                        'examiners_psc_rel',
+                        array(
+                         'examinerID'       => $examinerID,
+                         'centerID'         => $site,
+                         'active'           => 'Y',
+                         'pending_approval' => 'Y',
+                        )
+                    );
+                }
+            }
+        }
+        // Show success message even if email already exists for security reasons
+        $values['success'] = true;
+        return $values;
+    }
+
+    /**
+     * Login account.
```

Comment:
This should be more descriptive. Something like "Processes login data submitting via form, attempts authentication and verifies that a password has not expired.